tybalt/data_models.py

```diff
@@ -261,17 +270,25 @@ def nn(self, n_components, model='tybalt', transform_df=False, **kwargs):
                                    learning_rate=learning_rate,
                                    loss=loss,
                                    verbose=verbose,
+                                   tied_weights=tied_weights,
                                    optimizer=adage_optimizer)
             self.adage_fit.initialize_model()
             self.adage_fit.train_adage(train_df=self.nn_train_df,
                                        test_df=self.nn_test_df,
                                        adage_comparable_loss=adage_comp_loss)
-            self.adage_decoder_w = self.adage_fit.get_decoder_weights()
+            self.adage_decoder_w = (
+                self.adage_fit.get_weights(decoder=use_decoder_weights)
+                )
 
             features = ['dae_{}'.format(x) for x in range(0, latent_dim)]
-            self.adage_weights = pd.DataFrame(self.adage_decoder_w[1][0],
-                                              columns=self.df.columns,
-                                              index=features)
+            if use_decoder_weights:
+                self.adage_weights = pd.DataFrame(self.adage_decoder_w[1][0],
+                                                  columns=self.df.columns,
+                                                  index=features)
+            else:
+                self.adage_weights = pd.DataFrame(self.adage_decoder_w[1][0],
+                                                  index=self.df.columns,
+                                                  columns=features).T
```

Comment:
Why is the dataframe transposed here and not above?

Reply:
The "decoder" weights used in a `tied_weights` ADAGE model are the transposed encoder weights. To get it back into a format we like to deal with, a transpose is necessary. I agree this was confusing, so I moved the options from `tybalt/data_models.py` (6b4dc0f) to `tybalt/utils/base.py` (9e4bf02). I think it should belong in base.py.